packages/core/src/stateUtils.ts

```diff
@@ -333,9 +346,25 @@ export function formatTransition<
       `State "${stateNode.id}" has declared \`cond\` for one of its transitions. This property has been renamed to \`guard\`. Please update your code.`
     );
   }
+
+  const convertAction = (
+    action: UnknownAction,
+    i: number
+  ): UnknownActionObject => {
+    if (typeof action === 'string') {
+      return { type: action };
+    }
+    if (typeof action === 'function' && !('resolve' in action)) {
+      const type = `${stateNode.id}|${descriptor}:${index}:${i}`;
```

Comment:
is index necessary in this `type`? shouldn't `i` already be enough? for the record: I'm not sure if it's needed or not but im curious ;p